pkg/operator/apis/aro.openshift.io/v1alpha1/cluster_types.go

```diff
@@ -41,23 +41,26 @@ type InternetCheckerSpec struct {
 // ClusterSpec defines the desired state of Cluster
 type ClusterSpec struct {
 	// ResourceID is the Azure resourceId of the cluster
-	ResourceID      string              `json:"resourceId,omitempty"`
-	Domain          string              `json:"domain,omitempty"`
-	ACRDomain       string              `json:"acrDomain,omitempty"`
-	AZEnvironment   string              `json:"azEnvironment,omitempty"`
-	Location        string              `json:"location,omitempty"`
-	GenevaLogging   GenevaLoggingSpec   `json:"genevaLogging,omitempty"`
-	InternetChecker InternetCheckerSpec `json:"internetChecker,omitempty"`
-	VnetID          string              `json:"vnetId,omitempty"`
-	APIIntIP        string              `json:"apiIntIP,omitempty"`
-	IngressIP       string              `json:"ingressIP,omitempty"`
+	ResourceID          string              `json:"resourceId,omitempty"`
+	Domain              string              `json:"domain,omitempty"`
+	ACRDomain           string              `json:"acrDomain,omitempty"`
+	AZEnvironment       string              `json:"azEnvironment,omitempty"`
+	Location            string              `json:"location,omitempty"`
+	InfraID             string              `json:"infraId,omitempty"`
+	ArchitectureVersion int                 `json:"architectureVersion,omitempty"`
+	GenevaLogging       GenevaLoggingSpec   `json:"genevaLogging,omitempty"`
+	InternetChecker     InternetCheckerSpec `json:"internetChecker,omitempty"`
+	VnetID              string              `json:"vnetId,omitempty"`
+	APIIntIP            string              `json:"apiIntIP,omitempty"`
+	IngressIP           string              `json:"ingressIP,omitempty"`
 
 	Features FeaturesSpec `json:"features,omitempty"`
 }
 
 // FeaturesSpec defines ARO operator feature gates
 type FeaturesSpec struct {
 	PersistentPrometheus bool `json:"persistentPrometheus,omitempty"`
+	ReconcileNSGs        bool `json:"reconcileNSGs,omitempty"`
```

Comment:
False is default, and I don't want to have a hard override for now. So that we can enable it one by one on the clusters.
If it works we can force it to true (or remove the feature flag altogether if it's rolled out to 100% of clusters).